roles/servicetelemetry/tasks/component_qdr.yml

```diff
@@ -1,4 +1,147 @@
+        kind: Certificate
+        metadata:
+          name: "{{ meta.name }}-interconnect-openstack-credentials"
+          namespace: "{{ meta.namespace }}"
+        spec:
+          commonName: "{{ meta.name }}-interconnect"
+          dnsNames:
+            - "{{ meta.name }}-interconnect.{{ meta.namespace }}.svc.cluster.local"
+          issuerRef:
+            name: "{{ meta.name }}-interconnect-ca"
+          secretName: "{{ meta.name }}-interconnect-openstack-credentials"
+
+  - name: Create inter-router CA issuer
+    k8s:
+      state: '{{ "present" if servicetelemetry_vars.transports.qdr.enabled else "absent" }}'
+      definition:
+        apiVersion: cert-manager.io/v1
+        kind: Issuer
+        metadata:
+          name: "{{ meta.name }}-interconnect-inter-router-ca"
+          namespace: "{{ meta.namespace }}"
+        spec:
+          ca:
+            secretName: "{{ meta.name }}-interconnect-inter-router-ca"
+
+  - name: Create inter-router CA certificate
+    k8s:
+      state: '{{ "present" if servicetelemetry_vars.transports.qdr.enabled else "absent" }}'
+      definition:
+        apiVersion: cert-manager.io/v1
+        kind: Certificate
+        metadata:
+          name: "{{ meta.name }}-interconnect-inter-router-ca"
+          namespace: "{{ meta.namespace }}"
+        spec:
+          commonName: "{{ meta.name }}-interconnect-inter-router-ca"
+          isCA: true
+          issuerRef:
+            name: "{{ meta.name }}-interconnect-selfsigned"
+          secretName: "{{ meta.name }}-interconnect-inter-router-ca"
+
+  - name: Create inter-router credentials certificate
+    k8s:
+      state: '{{ "present" if servicetelemetry_vars.transports.qdr.enabled else "absent" }}'
+      definition:
+        apiVersion: cert-manager.io/v1
+        kind: Certificate
+        metadata:
+          name: "{{ meta.name }}-interconnect-inter-router-credentials"
+          namespace: "{{ meta.namespace }}"
+        spec:
+          commonName: "{{ meta.name }}-interconnect"
+          dnsNames:
+            - "{{ meta.name }}-interconnect.{{ meta.namespace }}.svc.cluster.local"
+          issuerRef:
+            name: "{{ meta.name }}-interconnect-inter-router-ca"
+          secretName: "{{ meta.name }}-interconnect-inter-router-credentials"
+  when: interconnect_manifest is not defined
+
+- name: Create Interconnect SASL ConfigMap
+  k8s:
+    state: '{{ "present" if servicetelemetry_vars.transports.qdr.enabled else "absent" }}'
+    definition:
+      kind: ConfigMap
+      apiVersion: v1
+      metadata:
+        name: "{{ meta.name }}-interconnect-sasl-config"
+        namespace: "{{ meta.namespace }}"
+        labels:
+          application: "{{ meta.name }}-interconnect"
+          interconnect_cr: "{{ meta-name }}-interconnect"
+      data:
+        qdrouterd.conf: |
+          pwcheck_method: auxprop
+          auxprop_plugin: sasldb
+          sasldb_path: /tmp/qdrouterd.sasldb
+  when: interconnect_manifest is not defined
```

Comment:
@csibbitt could I get another set of eyes to review if this makes? This is doing exactly what the Interconnect router results in through it's requests to AMQ Cert Manager, so this matches 1:1 what would have happened on STF 1.4 and OCP 4.8.

However now is our chance to adjust anything we think might look less than ideal. I keep looking at this sequence and something feels off to me. If I go through this, it seems like...

self-signed issuer references:
- selfsigned-ca cert
- openstack-ca cert
- inter-router-ca cert

interconnect-ca issuer references:
- ca-secret: selfsigned-ca cert
-- creates: openstack-credentials cert

inter-router-ca issuer references:
- ca-secret: inter-router-ca cert
-- creates: inter-router-credentials cert

So it seems that the `openstack-ca` certificate is never used. 

The openstack credentials certificate uses the interconnect-ca issuer.
The inter-router credentials certificate uses the inter-router-ca- issuer.

If we were going to clean this up, it seems like we'd create an openstack-ca issuer that used `ca.secretName` of `openstack-ca` certificate, then create the openstack-credentials certificate with an `issuerRef: ...-openstack-ca`.

How close am I? :)

Reply:
The interconnect-openstack-ca cert is "used" by the edge-listener L36 / 179 `sslProfile: openstack` which is defined on L51 / 194 and uses that cert. We'd need to experiment to figure out whether operationally this is "really used for anything", but AFAICT it's really just a "dummy" in this case (more below...)

I followed through your breakdown and it otherwise reads correctly, though I don't agree with your suggestion about the openstack-ca Issuer. I don't think it would make anything cleaner or confer any other advantage. Hopefully I can clarify more below...

TLDR; I don't think we really need to make any changes here, but some cleanup is likely possible.

To help me get this straight in my head I've cut down the code and arranged each certificate under it's issuer, similar to what you did above but I find it slightly easier to read.

```
kind: Issuer + Certificate (uses cert issued by self)
    name: "{{ meta.name }}-interconnect-selfsigned" 
kind: Certificate
    name: "{{ meta.name }}-interconnect-openstack-ca"
kind: Certificate
    name: "{{ meta.name }}-interconnect-inter-router-ca"

kind: Issuer (uses the same self-signed cert as above)
    name: "{{ meta.name }}-interconnect-ca" 
kind: Certificate
    name: "{{ meta.name }}-interconnect-openstack-credentials"

kind: Issuer (uses own cert issued by interconnect-selfsigned)
    name: "{{ meta.name }}-interconnect-inter-router-ca" 
kind: Certificate
    name: "{{ meta.name }}-interconnect-inter-router-credentials"      
```

There is one self-signed certificate in the system, `interconnect-selfsigned` which serves as the root of all trust. For some reason it's used to sign certs under two different "Issuers", itself and the `interconnect-ca` Issuer, possibly meant to be an "Intermediate CA", but usually an Intermediate CA has it's own cert signed by the root (like the inter-router one does). I'm not exactly sure what happened here, but it doesn't seem like a serious problem, and I think it MAY have been done this way to help simplify the OSP setup, see more below...

The `interconnect-ca` Isuer issues an `openstack-credentials` certificate which is used in the edge-listener's SSL profile. This cert is presented to clients that connect. You can check by comparing the outputs: 
* `oc get secret default-interconnect-openstack-credentials -o json | jq -r '.data["tls.crt"]' | base64 -d | openssl x509 -in - -text | less`
* `openssl s_client -servername default-interconnect-5671-stf15.apps.yourcluster -connect default-interconnect-5671-stf15.apps.yourcluster:443 2>/dev/null | openssl x509 -text | less`

The self-signed cert is also used to issue a cert to the more conventional looking (IMO) Intermediate CA called `interconnect-inter-router-ca` which is then used to generate and validate credentials for TLS mutual-auth connections between mesh members. Easy enough, very normal looking. Self-contained mesh-mode stuff. This should not change.

The self-signed cert is ALSO used to sign the `interconnect-openstack-ca` cert, which seems a bit weird to me. This is used by the edge-listener to validate client certificates presented during TLS mutual-auth, which is not something we currently do. I expect it might not work out of the box with a generated cert like this, anyways. I think in the TLS mutual-auth scenario we'd import this certificate from OSP, or from whatever CA issued the certs that OSP clients will present, rather than asking for it to be generated. AFAICT (haven't experimented yet), this cert looks like filler (since we don't use mutual-auth). We need to give the listener "something", but it's possible we could remove this cert and tell it to use `interconnect-openstack-credentials` as both the "caCert" and the "credentials" instead. I'm not sure if that's any less confusing, but it matches an example from the QDR docs so maybe it's not that crazy? https://access.redhat.com/documentation/en-us/red_hat_amq/7.6/html-single/deploying_amq_interconnect_on_openshift/index#creating-inter-cluster-router-network-router-ocp (step 6)

Finally, we export that self-signed certificate to OSP so it can verify the edge-listener's ceritificate (`openstack-credentials`) when it connects. This should not change.

So based on this, the only "Improvements" I can think to make are the following:

* Create an additional `interconnect-ca` Certificate issued by `interconnect-selfsigned`, to be used by the `interconnect-ca` Issuer so that it looks more like a normal intermediate CA 
  * This would likely alter our OSP connection procedures, as we'd either need to provide this cert to OSP at config-time, or have it bundled into the `interconnect-openstack-credentials` cert that the listener presents (either one of these steps would let openstack validate the credentials cert).
  * It's possible there is some complication here that lead to the same cert intentionally being used by the both the root and "intermediate" CAs
* (Or, better) Eliminate `interconnect-ca` entirely and issue `openstack-credentials` from the root. Everything should continue to work "as is" in this case AFAICT, just with one less Issuer to confuse us
* Rename `interconnect-openstack-ca` to `dummy-interconnect-openstack-ca`
  * (Or, better?) eliminate it entirely

Since this stuff is quite confusing to look at, eliminating the superfluous parts to help simplify things might be worth a try.